containers/src/Kokkos_ScatterView.hpp

```diff
@@ -209,8 +209,20 @@ struct ScatterValue<ValueType, Kokkos::Experimental::ScatterSum, DeviceType,
   KOKKOS_FORCEINLINE_FUNCTION void operator+=(ValueType const& rhs) {
     this->join(this->reference(), rhs);
   }
+  KOKKOS_FORCEINLINE_FUNCTION void operator++() {
+    this->join(this->reference(), 1);
+  }
+  KOKKOS_FORCEINLINE_FUNCTION void operator++(int) {
+    this->join(this->reference(), 1);
+  }
   KOKKOS_FORCEINLINE_FUNCTION void operator-=(ValueType const& rhs) {
-    this->join(this->reference(), -rhs);
+    this->join(this->reference(), ValueType(-rhs));
```

Comment:
this is wrong. If you do `a(i)-=unsigned(1);` you will get the wrong behavior.

Reply:
Can you please elaborate?

The arguments of `join()` have the same type anyway. Hence, an implicit conversion would happen anyway. Casting is only necessary for `operator--` to avoid warnings but I thought about adding them here as well for consistency.

Are you saying that something else should be checked in the tests?
AFAICT, we get, e.g.,
```
unsigned(2)-unsigned(10)==unsigned(2)+unsigned(-10)==unsigned(2)+unsigned(4294967286)==unsigned(4294967288)==int(-8)
```